Cause lists

actlikewill opened the pull request:
![image](https://github.com/user-attachments/assets/b2e6750e-79e6-4bac-80d3-d2105933a721)

closes https://github.com/laws-africa/peachjam/issues/2094

actlikewill commented on peachjam/models/judgment.py:
```diff
@@ -539,3 +539,21 @@ def __str__(self):
+    judges = models.ManyToManyField(Judge, blank=True, verbose_name=_("judges"))
+
+    def pre_save(self):
+        self.frbr_uri_doctype = "doc"
+        self.doc_type = "causelist"
```
@longhotsummer the frbr_uri  is not getting assigned correctly, should it be `/akn/doc/causelist/causelist/etc`? should the court be included?

longhotsummer commented on peachjam/models/judgment.py:
```diff
@@ -115,6 +115,23 @@ def save(self, *args, **kwargs):
+
+    def save(self, *args, **kwargs):
+        if not self.code:
+            self.code = f"{slugify(self.name)}"
```
```suggestion
            self.code = slugify(self.name)"
```

actlikewill replied:
fixed

actlikewill commented:
@longhotsummer I've added the document nature to the list filters and fixed the causelist titles to use the court registry by default and show the date in full
![image](https://github.com/user-attachments/assets/50776af6-66c2-4594-b10f-da369343ce2f)
